### PR TITLE
feat(ci): Github Workflow for Golang static checks and linting

### PR DESCRIPTION
**Pull request by dvjsharma**
## Description
Added Github Workflow for Golang static checks and linting
completes #33 

Note: To be merged once the codebase is error-free. following are the current errors:

Screenshot from 2024-01-11 17-25-33

Merge after #38 

CC @avinal @GMishx

### CODE REVIEW

File: .github/workflows/golangci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
+        check-latest: true
+        cache: true
+    
+    - name: lint
+      uses: golangci/golangci-lint-action@v3
+      with:
+          version: v1.51
```
Comment (GMishx) [inline .github/workflows/golangci.yml:33]: Please add a blank line at the bottom.

**Comment by dvjsharma** [comment]
Tested on a remote instance, working as expected after merge of #38 

image

**Comment by GMishx** [comment]
Hey @dvjsharma , since #38 is merged, please rebase the branch.

Also, the repo requires signed commits. Please sign the commit before pushing.

**Comment by dvjsharma** [comment]
Hi @GMishx, I've completed the rebase and signed the commit. The check has passed successfully. PTAL

**Review by GMishx — APPROVED** [review]
LGTM

**Comment by GMishx** [comment]
I don't see the commit is signed
image

It should have the Verified tag like this image

File: .github/workflows/golangci.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
+        check-latest: true
+        cache: true
+
+    - name: lint
+      uses: golangci/golangci-lint-action@v3
+      with:
+          version: v1.51
```
Comment (avinal) [inline .github/workflows/golangci.yml:33]:
```suggestion
          version: v1.55
```
We should use the latest version of a tool for first installation, unless there is a good reason not to.

**Review by avinal — CHANGES_REQUESTED** [review]
The rest looks good to me. FYI, we can still customize this using a YAML config. 

https://github.com/golangci/golangci-lint-action/blob/3efd423e8d927ea4a8725c50c8ab863118547ad5/README.md?plain=1#L57-L61

File: .github/workflows/golangci.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
```
[the remark this answers is not in the document]
Comment (dvjsharma) [reply .github/workflows/golangci.yml]: Done

File: .github/workflows/golangci.yml (hunk 2 above)
Comment (dvjsharma) [reply .github/workflows/golangci.yml]: Sure, I was just about to inquire about version updates. I've done some research, and v1.55 looks good to me.

**Comment by dvjsharma** [comment]
Hi @avinal, I've made the requested changes for the commit signature, tool version, and newline. PTAL.

P.S. Sorry for the delay i wasn't familiar about GPG Keys so it took time to configure :)

CC @GMishx

**Pull request merged**